### PR TITLE
CHAD-12648 SmartSense MultiPurpose sensor missing negative temperature handling

### PR DESCRIPTION
**Pull request by wkhenon**
Original PR: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1227

This PR incorporates the changes made in the original PR and also adds unit tests for these changes

### CODE REVIEW

File: drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua [hunk 1]
```diff
@@ -290,6 +290,48 @@ test.register_coroutine_test(
...
+      build_multi_status_report_message(mock_device, "\x04\x01\x00\x48")
+    })
+    test.socket.capability:__set_channel_ordering("relaxed")
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.temperatureMeasurement.temperature({ value = -25.5, unit = "C" })))
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.accelerationSensor.acceleration.active()))
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.contactSensor.contact.closed()))
+    test.socket.capability:__expect_send(mock_device:generate_test_message("main", capabilities.battery.battery(97)))
+  end)
+
+test.register_coroutine_test(
```
Comment (tpmanley) [inline drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:307]: I suggest testing different temperatures in each of these new tests. The handling of acceleration, contact and battery is tested by other tests so it'd be more valuable here is to test the edge cases for temp handling.

Comment (wkhenon) [reply drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:308]: added more tests with different negative values (31500af800ee206a8382804c95f8d55d66a39d53)

File: drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua [hunk 2]
```diff
@@ -290,6 +290,51 @@ test.register_coroutine_test(
     end
 )
 
+test.register_coroutine_test(
+  "Report from cluster 0xFC03, command 0x09 should be handled as: Temperature (-25.5 C)",
+  function()
+    test.socket.zigbee:__queue_receive({
+      mock_device.id,
+      build_multi_status_report_message(mock_device, "\x01\xFF\x00\x00")
```
Comment (tpmanley) [inline drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:298]: I'd suggest adding a test or changing one of these tests to also check the handling of 0xFFFF

Comment (greens) [reply drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:298]: is there something specific about 0xFFFF?

Comment (tpmanley) [reply drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:298]: It's an edge case

Comment (wkhenon) [reply drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:298]: handling in 0feff872d5d176eba6cdcec5f8a4e6f3b719c12f

Comment (tpmanley) [reply drivers/SmartThings/zigbee-contact/src/test/test_smartsense_multi.lua:298]: Thanks!

**Review by tpmanley — APPROVED** [review]
